rpc/src/rpc_subscriptions.rs

```diff
@@ -2780,6 +2781,91 @@ pub(crate) mod tests {
     }
 
+    #[test]
+    fn test_logs_subscribe() {
+        let GenesisConfigInfo {
+            genesis_config,
+            mint_keypair,
+            ..
+        } = create_genesis_config(100);
+        let bank = Bank::new_for_tests(&genesis_config);
+        let blockhash = bank.last_blockhash();
+        let bank_forks = Arc::new(RwLock::new(BankForks::new(bank)));
+
+        let alice = Keypair::new();
+        let bob = Keypair::new();
+
+        let tx = system_transaction::create_account(
+            &mint_keypair,
+            &alice,
+            blockhash,
+            1,
+            0,
+            &system_program::id(),
+        );
+
+        bank_forks
+            .read()
+            .unwrap()
+            .get(0)
+            .unwrap()
+            .process_transaction(&tx)
+            .unwrap();
+
+        let exit = Arc::new(AtomicBool::new(false));
+        let max_complete_transaction_status_slot = Arc::new(AtomicU64::default());
+        let optimistically_confirmed_bank =
+            OptimisticallyConfirmedBank::locked_from_bank_forks_root(&bank_forks);
+        let subscriptions = Arc::new(RpcSubscriptions::new_for_tests(
+            &exit,
+            max_complete_transaction_status_slot,
+            bank_forks,
+            Arc::new(RwLock::new(BlockCommitmentCache::new_for_tests())),
+            optimistically_confirmed_bank,
+        ));
+
+        let sub_config = RpcTransactionLogsConfig {
+            commitment: Some(CommitmentConfig::processed()),
+        };
+
+        let (rpc_all, mut receiver_all) = rpc_pubsub_service::test_connection(&subscriptions);
+        let sub_id_for_all = rpc_all
+            .logs_subscribe(RpcTransactionLogsFilter::All, Some(sub_config.clone()))
+            .unwrap();
+        assert!(subscriptions.control.logs_subscribed(None));
+
+        let (rpc_alice, mut receiver_alice) = rpc_pubsub_service::test_connection(&subscriptions);
+        let sub_id_for_alice = rpc_alice
+            .logs_subscribe(
+                RpcTransactionLogsFilter::Mentions(vec![alice.pubkey().to_string()]),
+                Some(sub_config.clone()),
+            )
+            .unwrap();
+        assert!(subscriptions.control.logs_subscribed(Some(&alice.pubkey())));
+
+        let (rpc_bob, mut receiver_bob) = rpc_pubsub_service::test_connection(&subscriptions);
+        let sub_id_for_bob = rpc_bob
+            .logs_subscribe(
+                RpcTransactionLogsFilter::Mentions(vec![bob.pubkey().to_string()]),
+                Some(sub_config.clone()),
+            )
+            .unwrap();
+        assert!(subscriptions.control.logs_subscribed(Some(&bob.pubkey())));
+
+        subscriptions.notify_subscribers(CommitmentSlots::default());
+
+        /**
+         * FIXME: These all timeout.
+         */
+        let response_all = receiver_all.recv();
+        let response_alice = receiver_alice.recv();
+        let response_bob = receiver_bob.recv();
```

Comment:
Can someone help me make these tests work? I gave it the 'ol college try, but these `recv()` calls all time out. There's something I don't understand about the banks test harness.